portal/static/js/src/empro.js

```diff
@@ -1,204 +1,546 @@
+  });
+};
+emproObj.prototype.hasSelectedOptOutDomains = function () {
+  return (
+    EmproObj.selectedOptOutDomains && EmproObj.selectedOptOutDomains.length
+  );
+};
+emproObj.prototype.setOptoutSubmitData = function () {
+  if (!EmproObj.hasSelectedOptOutDomains()) {
+    return;
+  }
+  var triggerObject = {};
+  EmproObj.selectedOptOutDomains.forEach((item) => {
+    triggerObject[item] = {
+      _opt_out_next_visit: "true",
+    };
+  });
+  var submitData = {
+    user_id: EmproObj.userId,
+    visit_month: EmproObj.visitMonth,
+    triggers: {
+      domains: triggerObject,
+    },
+  };
+  console.log("Data to be submitted for optout: ", submitData);
+  EmproObj.optOutSubmitData = submitData;
+};
+emproObj.prototype.setError = function (strSeletor, text) {
+  if (!document.querySelector(strSeletor)) return;
+  document.querySelector(strSeletor).innerText = text;
 };
-emproObj.prototype.initReportLink = function() {
-    if (!$("#emproModal").length) {
-        return;
+emproObj.prototype.setButtonsDisableState = function (
+  strSelector,
+  boolDisable
+) {
+  $(strSelector).attr("disabled", boolDisable);
+};
+emproObj.prototype.toggleSavingIndicator = function (strSelector, boolShow) {
+  if (boolShow) {
+    $(strSelector).removeClass("hide");
+    return;
+  }
+  $(strSelector).addClass("hide");
+};
+emproObj.prototype.setOptoutError = function (text) {
+  EmproObj.setError("#emproOptOutModal .error-message", text);
+};
+emproObj.prototype.setOptoutButtonsState = function (boolDisable) {
+  EmproObj.setButtonsDisableState(
+    "#emproOptOutModal .btn-submit, #emproOptOutModal .btn-dismiss",
+    boolDisable
+  );
+};
+emproObj.prototype.toggleOptoutSavingIndicator = function (boolShow) {
+  EmproObj.toggleSavingIndicator(
+    "#emproOptOutModal .saving-indicator-container",
+    boolShow
+  );
+};
+emproObj.prototype.onBeforeSubmitOptoutData = function () {
+  // reset error
+  EmproObj.setOptoutError("");
+  // set data in the correct format for submission to API
+  EmproObj.setOptoutSubmitData();
+  // disable buttons
+  EmproObj.setOptoutButtonsState(true);
+  // display saving in progress indicator icon
+  EmproObj.toggleOptoutSavingIndicator(true);
+};
+emproObj.prototype.onAfterSubmitOptoutData = function (data) {
+  // enable buttons
+  EmproObj.setOptoutButtonsState(false);
+  // hide saving in progress indicator icon
+  EmproObj.toggleOptoutSavingIndicator(false);
+  // show error if any
+  if (data && data.error) {
+    EmproObj.setOptoutError("System error: Unable to save your choices.");
+    return false;
+  }
+  EmproObj.submittedOptOutDomains = EmproObj.selectedOptOutDomains;
+  EmproObj.populateSelectedOptoutUI();
+  EmproObj.initOptOutModal(false);
+  EmproObj.initThankyouModal(true);
+  return true;
+};
+emproObj.prototype.handleSubmitOptoutData = function () {
+  if (!EmproObj.hasSelectedOptOutDomains()) {
+    EmproObj.setOptoutError(
+      "Please check at least one checkbox. Otherwise click 'Continue'."
```

Comment:
Is this referring to the button labelled "DISMISS", not "Continue"?

Reply:
thank you @mcjustin.  I have fixed the text now so when the user didn't select a checkbox, this is what will show:
![Error](https://github.com/uwcirg/truenth-portal/assets/12942714/15774973-3927-4b55-ae6d-8b5142e4ba74)

Does the "Continue" button seem redundant? If so, I will remove.

Reply:
It seems redundant. Thanks for this screenshot... stepping back a bit, I don't recall that they should check at least one checkbox. Do you see that documented anywhere? 
If it's not a requirement, then if they click DISMISS and they haven't selected anything yet, perhaps we should show text that says "Are you certain that you don't want to select anything? If so, click DISMISS again."

Reply:
@mcjustin right, I don't recall either, that it is a requirement.  I made the changes and attached is the latest screenshot after the user clicks DISMISS and hasn't selected anything:
![dismiss](https://github.com/uwcirg/truenth-portal/assets/12942714/29233eb6-4b8d-4957-a6a6-6fd2b7aaee29)

Reply:
Looks good to me, thanks Amy!